inc/classes/class-srm-post-type.php

```diff
@@ -671,18 +673,65 @@ public static function factory() {
 	 */
 	public function load_resources() {
 		if ( 'redirect_rule' === get_post_type() ) {
+			wp_enqueue_style( 'redirectjs', plugin_dir_url( 'safe-redirect-manager/safe-redirect-manager.php' ) . 'assets/css/redirect.css', array(), SRM_VERSION );
 			wp_enqueue_script( 'redirectjs', plugin_dir_url( 'safe-redirect-manager/safe-redirect-manager.php' ) . 'assets/js/redirect.js', array( 'jquery' ), SRM_VERSION );
 			wp_localize_script(
 				'redirectjs',
 				'redirectValidation',
 				array(
 					'urlError' => __( 'There are some issues validating the URL. Please try again.', 'safe-redirect-manager' ),
 					'fail'     => __( 'There is an existing redirect with the same Redirect From URL. You may <a href="%s">Edit</a> the redirect or try other `from` URL.', 'safe-redirect-manager' ),
+					'ajax_url'   => admin_url( 'admin-ajax.php' ),
+					'ajax_nonce' => wp_create_nonce( 'srm_autocomplete_nonce' ),
 				)
 			);
 		}
 	}
 
+	/**
+	 * Fetches posts and pages based on the 'Redirect to:' field value.
+	 * This function handles an AJAX request, returning a JSON array of posts
+	 * and pages that match the search term.
+	 *
+	 * @return void
+	 */
+	public function srm_autocomplete() {
+		check_ajax_referer( 'srm_autocomplete_nonce', 'security' );
```

Comment:
@bmarshall511 Can we also check user capability, whether the user can read/search other post types?

Reply:
Definitely! I've updated the code to include a check for the `read` capability.

Reply:
@bmarshall511 Can we use [`srm_manage_redirects`](https://github.com/10up/safe-redirect-manager/blob/832ca3411801a4768772410d8dc8284457f9600d/inc/classes/class-srm-post-type.php#L469)?

Reply:
Sure! Updated accordingly.

Reply:
@bmarshall511 pardon me 🙏 . My last comment was for something other than renaming the nonce action name. [srm_manage_redirects](https://github.com/10up/safe-redirect-manager/blob/832ca3411801a4768772410d8dc8284457f9600d/inc/classes/class-srm-post-type.php#L469) is a capability and we can check for this capability instead of `read`. 

Let me know if you have questions.